sympy/combinatorics/perm_groups.py

```diff
@@ -694,6 +694,70 @@ def basic_transversals(self):
             self.schreier_sims()
         return self._transversals
 
+    def composition_series(self):
+        r"""
+        Return the composition series for a group as a list
+        of permutation groups.
+
+        The composition series for a group `G` is defined as a
+        subnormal series `G = H_0 > H_1 > H_2 \ldots` A composition
+        series is a subnormal series such that each factor group
+        `H(i+1) / H(i)` is simple.
+        A subnormal series is a composition series only if it is of
+        maximum length.
+
+        Examples
+        ========
+        >>> from sympy.combinatorics.named_groups import SymmetricGroup
+        >>> from sympy.combinatorics.named_groups import CyclicGroup
+        >>> S = SymmetricGroup(12)
+        >>> G = S.sylow_subgroup(2)
+        >>> C = G.composition_series()
+        >>> [H.order() for H in C]
+        [1024, 512, 256, 128, 64, 32, 16, 8, 4, 2, 1]
+        >>> G = S.sylow_subgroup(3)
+        >>> C = G.composition_series()
+        >>> [H.order() for H in C]
+        [243, 81, 27, 9, 3, 1]
+        >>> G = CyclicGroup(12)
+        >>> C = G.composition_series()
+        >>> [H.order() for H in C]
+        [12, 6, 3, 1]
+
+        """
+        if not self.is_solvable:
+            raise NotImplementedError('Group should be solvable')
+        der = self.derived_series()
+        series = [self]
+
+        for i in range(len(der)-1):
+            H = der[i+1]
+            Hgens = H.generators
+            seg = []
+            for g in der[i].generators:
+                if not g in H:
+                    order = g.order()
+                    for p, e in factorint(order).items():
+                        for j in range(e):
+                            if len(der) == 2:
+                                    seg.insert(0, PermutationGroup([g] + Hgens))
+                                    g = g**p
+                            else:
+                                g = g**(p**j)
+                                if not g in H:
```

Comment:
It should not be necessary to check `g in H`. That is relatively expensive and can be avoided if `order` is the least integer such that `g**order` is in `H`.